conflation/trace_fetching/mapillary.py

```diff
@@ -9,54 +9,81 @@
 )
+SEQUENCE_URL = "https://graph.mapillary.com/image_ids?fields=id&sequence_id={}&access_token={}"
+IMAGES_URL = "https://graph.mapillary.com/images?fields=captured_at,geometry&image_ids={}&access_token={}"
 
-SEQUENCE_URL = "https://a.mapillary.com/v3/sequences_without_images?client_id={}&bbox={}&per_page={}&start_date={}"
-IMAGES_URL = "https://a.mapillary.com/v3/images?client_id={}&sequence_keys={}&per_page={}"
-MAX_FILES_IN_DIR = 500  # Maximum number of files we will put in one directory
+# Mapillary v4 supports "coverage" search over a zoom level from 0 to 5. We perform the coverage search at zoom 5
+COVERAGE_ZOOM = 5
+# We then perform the actual search for trace sequences at zoom 14
+BBOX_SECTION_ZOOM = 14
 
 
 def run(
     bbox: str, traces_dir: str, tmp_dir: str, config: dict, processes: int, access_token: str
-) -> int:
+) -> None:
     """
     Entrypoint for pulling trace date from Mapillary APIs. Will pull all trace data in the given bbox and store it in
     the traces_dir, using the number of processes specified and any conf values from the `config` JSON.
 
     :param bbox: Bounding box we are searching over, in the format of 'min_lon,min_lat,max_lon,max_lat'
     :param traces_dir: Dir where trace data will be pickled to
     :param tmp_dir: Dir where temp output files will be stored (should be empty upon completion)
-    :param config: Dict of configs, see the .README or the conf param of make_trace_data_requests()
+    :param config: Dict of configs. Mandatory keys are ['client_id', 'client_secret']. Optional keys are
+        ['start_date', 'max_sequences_per_bbox_section', 'skip_if_fewer_imgs_than']
     :param processes: Number of threads to use
-    :param access_token: Mapillary access token for API calls
+    :param access_token: Mapillary v4 access token (obtained through OAuth)
     """
-    # Break the bbox into sections and save it to a pickle file
-    bbox_sections = split_bbox(traces_dir, bbox)
 
-    print("bbox_sections={}".format(bbox_sections))
-    print("access_token={}".format(access_token))
-    print("Rest is not yet implemented.")
-    return 0
+    # Requests session for persistent connections and timeout settings
+    session = requests.Session()
+    retry_strategy = Retry(
+        total=5, status_forcelist=[429, 500, 502, 503, 504], backoff_factor=3
+    )
+    adapter = HTTPAdapter(max_retries=retry_strategy)
+    session.mount("https://", adapter)
+    session.mount("http://", adapter)
+
+    # We only want to consider recent sequences, so we take `start_date` as an optional param, and only consider
+    # sequences dated past this given date
+    start_date = (
+        parser.parse(config["start_date"])
+        if "start_date" in config
+        else SEQUENCE_START_DATE_DEFAULT
+    )
+    # Convert to epoch milliseconds, since Mapillary's captured_at is in epoch milliseconds
+    start_date_epoch = (
+        start_date - datetime.datetime.utcfromtimestamp(0)
+    ).total_seconds() * 1000.0
```

Comment:
This stuff used to be done later / within `initialize_multiprocess` but I pulled it out here since it's necessary for `split_bbox` as well now (since we are making calls to the API and doing some processing).